src/crimson/common/operation.h

```diff
@@ -702,35 +693,31 @@ class OrderedConcurrentPhaseT : public PipelineStageIT<T> {
 
     void exit() final {
       if (barrier) {
-	static_cast<void>(
-	  std::move(*barrier).then([phase=this->phase] { phase->mutex.unlock(); }));
-	barrier = std::nullopt;
-	phase = nullptr;
-      }
-      if (phase) {
-	std::ignore = seastar::smp::submit_to(
```

Comment:
@athanatos Probably it's simpler to use `OrderedConcurrentPhaseT` only in the local reactor, and for the special cross-core case, relying on the newly introduced class `crosscore_t` .

Unlike the original `OrderedConcurrentPhaseT`, `crosscore_t` doesn't rely on any additional `submit_to()` to preserve the ordering.